Phylanx variant doesn't compile with MSVC 19.15.26726

Output
```
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\detail\is_placeholder.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\detail\is_placeholder.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_argument.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_argument.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\generate_transform_rules.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\generate_transform_rules.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\detail\is_identifier.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\detail\is_identifier.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\generate_ast.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\generate_ast.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\transform_ast.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\transform_ast.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\detail\tagged_id.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\detail\tagged_id.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\node.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\node.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\call_function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\call_function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\patterns.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\patterns.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\compiler\compiler.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\compiler\compiler.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\assert_condition.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\assert_condition.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice_range.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice_range.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\detail\is_placeholder_ellipses.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\detail\is_placeholder_ellipses.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_variable.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_variable.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\format_string.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\format_string.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\enable_tracing.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\enable_tracing.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\parser\expression.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\parser\expression.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\define_variable.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\define_variable.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\debug_output.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\debug_output.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\detail\is_placeholder.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\detail\is_function_call.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\detail\is_function_call.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\base_primitive.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\base_primitive.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\compile.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\compile.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\lambda.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\lambda.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_argument.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\generate_transform_rules.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\console_output.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\console_output.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\ast\detail\is_literal_value.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\ast\detail\is_literal_value.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice_node_data.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice_node_data.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\detail\is_identifier.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\generate_ast.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\transform_ast.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\detail\tagged_id.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\node.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\call_function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\patterns.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\compiler\compiler.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\assert_condition.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice_range.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\detail\is_placeholder_ellipses.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_variable.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\format_string.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\enable_tracing.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\parser\expression.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\define_variable.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\debug_output.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\detail\is_placeholder.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\detail\is_function_call.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\base_primitive.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\compile.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\lambda.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_argument.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\generate_transform_rules.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\console_output.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\ast\detail\is_literal_value.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(939): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice_node_data.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\detail\is_identifier.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\generate_ast.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\transform_ast.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\detail\tagged_id.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\node.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\call_function.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\patterns.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\compiler\compiler.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\assert_condition.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\slice_range.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\detail\is_placeholder_ellipses.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(2057): note: see reference to class template instantiation 'mpark::detail::impl<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\access_variable.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\format_string.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\enable_tracing.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\parser\expression.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\define_variable.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\debug_output.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(2057): note: see reference to class template instantiation 'mpark::detail::impl<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\detail\is_placeholder.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\ast\detail\is_function_call.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1758): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\base_primitive.cpp)
.
.
.
```

Update: Replacing [`variant.hpp`](https://github.com/STEllAR-GROUP/phylanx/blob/master/phylanx/util/variant.hpp) to [Michael Park's latest version](https://github.com/mpark/variant/releases/download/v1.3.0/variant.hpp) does not help.
```
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(935): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(935): note: failure was caused by a read of a variable outside its lifetime (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(935): note: see usage of '<traits_0>' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(1754): note: see reference to class template instantiation 'mpark::detail::traits<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(2053): note: see reference to class template instantiation 'mpark::detail::impl<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\ir\node_data.hpp(331): note: see reference to class template instantiation 'mpark::variant<double,blaze::DynamicVector<T,false>,blaze::DynamicMatrix<T,false>,blaze::CustomVector<T,true,true,false>,blaze::CustomMatrix<T,true,true,false>>' being compiled
1>        with
1>        [
1>            T=double
1>        ] (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\ir\node_data.hpp(338): note: see reference to class template instantiation 'phylanx::ir::node_data<double>' being compiled (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\vcpkg\installed\x64-windows\include\blaze\math\views\check.h(138): note: see reference to class template instantiation 'blaze::Check<false>' being compiled (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\vcpkg\installed\x64-windows\include\blaze\math\views\check.h(121): note: see reference to class template instantiation 'blaze::Check<true>' being compiled (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\hpx\hpx\parallel\execution_policy.hpp(675): note: see reference to class template instantiation 'hpx::parallel::execution::parallel_policy_executor<hpx::launch>' being compiled (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(963): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.15.26726\include\initializer_list(42): note: failure was caused by non-constant arguments or reference to a non-constant symbol (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.15.26726\include\initializer_list(42): note: see usage of '$S6' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(968): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.15.26726\include\initializer_list(42): note: failure was caused by non-constant arguments or reference to a non-constant symbol (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.15.26726\include\initializer_list(42): note: see usage of '$S8' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(973): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(973): note: failure was caused by a read of an uninitialized symbol (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(973): note: see usage of 'copy_constructible_trait' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(979): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(979): note: failure was caused by a read of an uninitialized symbol (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(979): note: see usage of 'move_constructible_trait' (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\repos\phylanx\phylanx\util\detail\variant.hpp(985): error C2131: expression did not evaluate to a constant (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
1>c:\program files (x86)\microsoft visual studio\2017\community\vc\tools\msvc\14.15.26726\include\initializer_list(42): note: failure was caused by non-constant arguments or reference to a non-constant symbol (compiling source file C:\Repos\phylanx\src\execution_tree\primitives\primitive_component_base.cpp)
.
.
.
```

--- a/phylanx/ir/dictionary.hpp
+++ b/phylanx/ir/dictionary.hpp
@@ -34,7 +34,7 @@ namespace std {
         using argument_type = phylanx::util::recursive_wrapper<
             phylanx::execution_tree::primitive_argument_type>;
         using result_type = std::size_t;
-        result_type operator()(argument_type const& s) const noexcept;
+        PHYLANX_EXPORT result_type operator()(argument_type const& s) const noexcept;
     };
 }
 

--- a/phylanx/util/detail/variant.hpp
+++ b/phylanx/util/detail/variant.hpp
@@ -242,7 +242,7 @@ namespace std {
 #define MPARK_TYPE_PACK_ELEMENT
 #endif
 
-#if defined(__cpp_constexpr) && __cpp_constexpr >= 201304
+#if defined(__cpp_constexpr) && __cpp_constexpr >= 201304 && !defined(_MSC_VER)
 #define MPARK_CPP14_CONSTEXPR
 #endif
 

